plugin/inc/partner.class.php

```diff
@@ -326,4 +326,25 @@ public function displayPreferenceData(): void
         echo "<li id='partner-title'$class>{$this->fields['name']} - {$this->customfields->fields['uic_field']}</li>";
     }
 
+    public static function ajaxRemoveSpacesFromName()
+    {
+        $id = IserviceToolBox::getInputVariable('id', null);
+        if (empty($id)) {
+            return _t('No IDs provided for removing spaces from names.');
+        }
+
+        $partner = new self();
+        if ($partner->getFromDB($id)) {
+            $name = trim(preg_replace('/\s+/', ' ', $partner->fields['name']));
+            if ($name !== $partner->fields['name']) {
+                $partner->fields['name'] = $name;
+                if ($partner->update(['id' => $id, 'name' => $name])) {
+                    return IserviceToolBox::RESPONSE_OK;
+                }
+            }
+        }
```

Comment:
The `CommonDBTM::update` includes a `getFromDB` and check if the values have indeed changed, so a single line is enough (same for the printer!):
```suggestion
        $partner = new self();
        if ($partner->update(['id' => $id, 'name' => trim(preg_replace('/\s+/', ' ', $partner->fields['name']))])) {
            return IserviceToolBox::RESPONSE_OK;
        }
```